pkg/service/tenant_test.go

```diff
@@ -797,6 +800,62 @@ func Test_tenantService_ListTenantMembers(t *testing.T) {
 			want:    &v1.ListTenantMembersResponse{},
 			wantErr: nil,
 		},
+		{
+			name: "indirect membership in multiple projects",
+			req: &v1.ListTenantMembersRequest{
+				TenantId:         "github",
+				IncludeInherited: pointer.Pointer(true),
+			},
+			prepare: func() {
+				err := tenantStore.Create(ctx, &v1.Tenant{Meta: &v1.Meta{Id: "github"}})
+				require.NoError(t, err)
+				err = tenantStore.Create(ctx, &v1.Tenant{Meta: &v1.Meta{Id: "azure"}})
+				require.NoError(t, err)
+				err = projectStore.Create(ctx, &v1.Project{Meta: &v1.Meta{Id: "1"}, TenantId: "github"})
+				require.NoError(t, err)
+				err = projectStore.Create(ctx, &v1.Project{Meta: &v1.Meta{Id: "2"}, TenantId: "github"})
+				require.NoError(t, err)
+				err = projectMemberStore.Create(ctx, &v1.ProjectMember{Meta: &v1.Meta{Annotations: map[string]string{"metalstack.cloud/project-role": "owner"}}, ProjectId: "1", TenantId: "github"})
+				require.NoError(t, err)
+				err = projectMemberStore.Create(ctx, &v1.ProjectMember{Meta: &v1.Meta{Annotations: map[string]string{"metalstack.cloud/project-role": "owner"}}, ProjectId: "2", TenantId: "github"})
+				require.NoError(t, err)
+				err = projectMemberStore.Create(ctx, &v1.ProjectMember{Meta: &v1.Meta{Annotations: map[string]string{"metalstack.cloud/project-role": "viewer"}}, ProjectId: "2", TenantId: "azure"})
+				require.NoError(t, err)
+				err = tenantMemberStore.Create(ctx, &v1.TenantMember{Meta: &v1.Meta{Annotations: map[string]string{"metalstack.cloud/tenant-role": "owner"}}, MemberId: "github", TenantId: "github"})
+				require.NoError(t, err)
```

Comment:
This project does not know about metalstack.cloud:

```suggestion
				err = projectMemberStore.Create(ctx, &v1.ProjectMember{Meta: &v1.Meta{Annotations: map[string]string{"project-role": "owner"}}, ProjectId: "1", TenantId: "github"})
				require.NoError(t, err)
				err = projectMemberStore.Create(ctx, &v1.ProjectMember{Meta: &v1.Meta{Annotations: map[string]string{"project-role": "owner"}}, ProjectId: "2", TenantId: "github"})
				require.NoError(t, err)
				err = projectMemberStore.Create(ctx, &v1.ProjectMember{Meta: &v1.Meta{Annotations: map[string]string{"project-role": "viewer"}}, ProjectId: "2", TenantId: "azure"})
				require.NoError(t, err)
				err = tenantMemberStore.Create(ctx, &v1.TenantMember{Meta: &v1.Meta{Annotations: map[string]string{"tenant-role": "owner"}}, MemberId: "github", TenantId: "github"})
				require.NoError(t, err)
```